api/xds_protocol.rst

```diff
@@ -469,3 +469,155 @@ occurred via a :ref:`resource update <xds_protocol_resource_update>`.
 
 .. |Multiple EDS requests on the same stream| image:: diagrams/eds-same-stream.svg
 .. |Multiple EDS requests on distinct streams| image:: diagrams/eds-distinct-stream.svg
+
+Implementations
+---------------
+
+.. _on_demand_vhds_rds_protocol:
+
+Virtual Host Discovery Service
```

Comment:
General comment: I wonder if this section would be better off placed into a separate document.  This doc is mostly about the xds protocol as a generic resource mechanism; it does not currently contain details about any of the specific xds APIs.

Reply:
This was originally in a separate document, but the suggestion was made in this PR to make it a child of the xDS documentation. Happy to do it either way.

Reply:
I think just a section on per-xDS caveats is fine. @markdroth I think you're asking for a distinction between transport protocol and data model here :-P

Reply:
I'm wondering whether it might be worth it to move a lot of this content over to the VHDS configuration page. I think a lot of it is useful (it's certainly enough for all of us to understand it and cause a lot of conversation here). 

Seems like the way that a lot of the other xDS protocol documentation works is just to describe ADS and then skip to the API endpoints/protobufs. Is it useful to explain how to use it? Or would it be better just to have a couple of high level bullets similar to the listeners page? 

https://www.envoyproxy.io/docs/envoy/v1.12.0/configuration/listeners/lds

Reply:
Refactored to remove this from the xds_protocol page and add to the VHDS configuration page. Let me know if that's better.

Reply:
This looks really good to me, but I'll let @htuch make the final call on this.